### PR TITLE
DOCSP-40574: Add Geospatial page

### PR DESCRIPTION
**Pull request by mcmorisi**
# Pull Request Info

Largely based off the [Go version](https://www.mongodb.com/docs/drivers/go/upcoming/fundamentals/geo/) of the page, with a little [Java version](https://www.mongodb.com/docs/drivers/java/sync/upcoming/fundamentals/crud/read-operations/geo/) sprinkled in.

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-40574>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/DOCSP-40574-geo/fundamentals/geo>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/geo.txt [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+:manual:`GeoJSON manual entry </reference/geojson/>`.
+
+For more information on the GeoJSON format, see the
+`official IETF specification <https://datatracker.ietf.org/doc/html/rfc7946>`__.
+
+Legacy Coordinate Pairs
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use legacy coordinate pairs to store data that represents geospatial information
+on a two-dimensional Euclidean plane.
```
Comment (caitlindavey) [inline source/fundamentals/geo.txt:133]:
```suggestion
on a two-dimensional plane.
```
See same comment as above

File: source/fundamentals/geo.txt [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+- ``2dsphere``, used for GeoJSON data
+- ``2d``, used for legacy coordinate pairs
+
+To learn more about how to create geospatial indexes, see the :ref:`geospatial-indexes`
+section of the Indexes guide.
+
+Query Operators
+~~~~~~~~~~~~~~~
+
+To query your geospatial data, use one of the following query operators:
```
Comment (caitlindavey) [inline source/fundamentals/geo.txt:157]:
```suggestion
To query geospatial data, use one of the following query operators:
```

File: source/fundamentals/geo.txt [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+Query by Polygon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example queries for documents with a ``location.geo`` field value that exists
+within the boundaries of Manhattan Island.
+
+.. code-block:: csharp
+
+   // Polygon representation of Manhattan Island
```
Comment (caitlindavey) [inline source/fundamentals/geo.txt:226]:
```suggestion
   // Polygon representation of Manhattan
```
Nit: Manhattan Island sounds stiff

File: source/fundamentals/geo.txt [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+   :copyable: False
+
+   { "_id" : ObjectId("59a47287cfa9a3a73e51e8e2"), "theaterId" : 1908 }
+   { "_id" : ObjectId("59a47286cfa9a3a73e51e838"), "theaterId" : 1448 }
+
+Query by Polygon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example queries for documents with a ``location.geo`` field value that exists
+within the boundaries of Manhattan Island.
```
Comment (caitlindavey) [inline source/fundamentals/geo.txt:222]:
```suggestion
within the boundaries of Manhattan.
```
Nit: Manhattan Island sounds stiff

**Review by caitlindavey — APPROVED** [review]
Some small tweaks but otherwise LGTM!

File: source/fundamentals/geo.txt [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to work with **geospatial data** formats,
```
Comment (BorisDog) [inline source/fundamentals/geo.txt:23]: I should be  "work with geospatial data, data formats ...".

File: source/fundamentals/geo.txt [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+
+The following is the position of the MongoDB Headquarters in New York City, NY.
+
+.. code-block:: csharp
+
+   GeoJson.Position(-73.986805, 40.7620853)
+
+.. important:: Longitude then Latitude
+
+  GeoJSON orders coordinates with longitude first and latitude second.
```
Comment (BorisDog) [inline source/fundamentals/geo.txt:72]: Consider also adding` GeoJson.Geographic(-73.986805, 40.7620853)` example, with ` GeoJson.Geographic` geographic coordinates names are used: longitude and longitude

File: source/fundamentals/geo.txt [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+The results of the preceding example contain the following documents:
+
+.. code-block:: json
+   :copyable: False
+
+   { "_id" : ObjectId("59a47287cfa9a3a73e51e8e2"), "theaterId" : 1908 }
+   { "_id" : ObjectId("59a47286cfa9a3a73e51e838"), "theaterId" : 1448 }
+
+Query by Polygon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (BorisDog) [inline source/fundamentals/geo.txt:218]: Should this be:

```
Query by Polygon
~~~~~~~~~~~~
```

?

**Review by BorisDog — CHANGES_REQUESTED** [review]
Few minor comments.

File: source/fundamentals/geo.txt (hunk 7 above)
Comment (mcmorisi) [reply source/fundamentals/geo.txt:218]: Good catch!

File: source/fundamentals/geo.txt (hunk 6 above)
Comment (mcmorisi) [reply source/fundamentals/geo.txt:72]: Could you provide a little more detail on this? A little unsure on how you would incorporate this example (also you say "longitude and longitude" here).

Comment (BorisDog) [reply source/fundamentals/geo.txt:72]: Sorry, lon and lon is a typo;

There is a method in our `GeoJson` class, `GeoJson.Geographic` for constructing coordinates with Longitude and Latitude properties. It's equivalent, to Position, just different naming for clarity. We could mentioned that as well.

**Review by BorisDog — APPROVED** [review]
LGTM

**Pull request merged**